azure-pipelines.yml

```diff
@@ -0,0 +1,86 @@
+trigger:
+  branches:
+    include:
+      - master
+      - refs/tags/*
+
+pr: ['*']
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+variables:
+  isMain: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
+  npm_config_cache: $(Pipeline.Workspace)/.npm
+
+stages:
+  - stage: Main
+    jobs:
+      - job: Build
+        steps:
+          - task: NodeTool@0
+            inputs:
+              versionSpec: 14.x
+            displayName: Install Node.js
+
+          - task: Cache@2
+            inputs:
+              key: 'npm | "$(Agent.OS)" | package-lock.json'
+              restoreKeys: |
+                npm | "$(Agent.OS)"
+              path: $(npm_config_cache)
+            displayName: Cache npm
+
+          - script: npm ci
+            displayName: Install dependencies
+
+          - script: |
+              npm run check:themes
+              npm run lint
+            displayName: Run linting
+
+          - script: |
+              npm run test:unit -- --coverage
+              bash <(curl -s https://codecov.io/bash)
+            displayName: Run unit tests
+
+          - script: |
+              npm run docs
+              npm run test:visual
+              argos upload test/screenshots/local -T "$ARGOS_TOKEN"
+            displayName: Run visual regression tests
+            condition: and(ne(variables('SAUCE_ACCESS_KEY'), ''), ne(variables('ARGOS_TOKEN'), ''))
+            env:
+              ARGOS_TOKEN: $(ARGOS_TOKEN)
+              SAUCE_ACCESS_KEY: $(SAUCE_ACCESS_KEY)
+
+  - stage: Release
+    condition: and(succeeded(), eq(variables.isMain, true))
+    jobs:
+      - job: Release
+        steps:
+          - script: |
+              git config --global user.email "$GIT_AUTHOR_EMAIL"
+              git config --global user.name "$GIT_AUTHOR_NAME"
+              git remote set-url origin "https://$GITHUB_TOKEN@github.com/$GITHUB_REPOSITORY.git" > /dev/null 2>&1
+              echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> .npmrc 2> /dev/null
+            env:
+              GITHUB_TOKEN: $(GITHUB_TOKEN)
+              NPM_TOKEN: $(NPM_TOKEN)
+            displayName: Configure services
+
+          - script: ASSET_PATH=/thema/ npm run docs
+            displayName: 'Build documentation site'
+
+          - script: npx semantic-release
+            displayName: 'Publish package'
+
+          - script: |
+              git fetch --all
+              git checkout --force --track origin-pages/gh-pages
+              git branch --set-upstream-to origin-pages/gh-pages
+              git reset --hard
+              git add -f --all docs
+              git commit --message "Deploy documentation to GitHub pages"
+              git push --quiet --force
+            displayName: 'Deploy to GitHub Pages'
```

Comment:
The `dev-config` set up now includes a plugin to publish `./docs` to GitHub pages. So potentially all of this could be simplified to:

```yaml
      - script: |
          npm run build
          ASSET_PATH=/thema/ npm run docs
          npx semantic-release
        env:
          GIT_AUTHOR_NAME: Stencila CI Bot
          GIT_AUTHOR_EMAIL: ci@stenci.la
          GIT_COMMITTER_NAME: Stencila CI Bot
          GIT_COMMITTER_EMAIL: ci@stenci.la
          GITHUB_TOKEN: $(GITHUB_TOKEN)
          NPM_TOKEN: $(NPM_TOKEN)
        displayName: Release package
```

See https://github.com/stencila/executa/blob/master/azure-pipelines.yml for example.